hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/container/keyvalue/statemachine/background/BlockDeletingService.java

```diff
@@ -198,8 +198,15 @@ private boolean isDeletionAllowed(ContainerData containerData,
       if (ozoneContainer.getWriteChannel() instanceof XceiverServerRatis) {
         XceiverServerRatis ratisServer =
             (XceiverServerRatis) ozoneContainer.getWriteChannel();
-        PipelineID pipelineID = PipelineID
-            .valueOf(UUID.fromString(containerData.getOriginPipelineId()));
+        PipelineID pipelineID;
```

Comment:
After some debugging, the issue is that `containerData.getOriginPipelineId()` is an empty string for EC Containers. I have not been able to track down how the pipelineID gets added to the `writeChunk` call for Ratis requests - I tried to find it in the debugger, but I suspect it happens inside the Xceiver Ratis code. Either way, I did see that the writeChunk command, which creates the container on the first write has a pipelineID set for Ratis writes, but not for EC writes. This kind of makes sense, as EC writes don't really care about the pipeline on a DN - they are all standalone writes.

The code here at line 198, will always end up returning true, even if its not a Ratis write. I ran into this before trying to do something else on the DN where I was trying to detect an EC or Ratis container.

All the logic inside this IF branch is related Ratis stuff - whether the pipeline still exists or not.

Would we be better adding a new IF branch, something like:

```
else if (containerData.getOriginPipelineId == emptyString or Null) {
  // This is not a ratis container so allow the delete
  return true;
} else {
  // existing code as it is
}
```

This would avoid the ugly exception handling, and I've been told catching exceptions like this is Java is relatively slow (ie avoid using exception handling for common cases), so if there are a lot of block deletes for EC, it may cause a performance problem as this exception will always get thrown.

Reply:
This change only checks empty or null pipeline id, which is sufficient for the EC case.

However, if there is a malformed container whose pipeline id is some random non-empty string,
should we be able to delete blocks in it?